### ISSUE TITLE
ui/tech-debt: JSX.Element has been deprecated

### ISSUE THREAD

**Issue by Forfold** [target/goalert#3435]
We'll need to rethink our component typings, as JSX as a whole has been deprecated in favor of React.JSX.

This is a good opportunity to decide if we want all of our components to be one of:
- React.JSX
- React.ReactNode (allows null, may be most preferred to work well with Suspense down the road)
- React.FC<Props>

![Screenshot 2023-11-08 at 8 45 37 AM](https://github.com/target/goalert/assets/11381794/32fdaf2a-e55b-41ef-93f8-c8503d674d82)

**Comment by mastercactapus** [comment]
I think `ReactNode` is the preferred way and what we intended JSX to be used for

**Comment by Forfold** [comment]
Agreed, I would be fine with moving towards ReactNode from the top down

**Comment by Kamlesh72** [comment]
@Forfold is this issue beginner friendly?

**Comment by Forfold** [comment]
Yes and no. It doesn't require a ton of effort, but it's a bit of a spiderweb in that I've noticed if I flip certain files to ReactNode from JSX.Element it freaks out saying it's not expected. So I think we need to start with the components at the highest level

**Comment by mastercactapus** [comment]
I think for this, we should just do `JSX.Element` -> `React.JSX.Element` for now and not tie it to changing to `ReactNode`

There's just too many usages to do it all at once.
